src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -111,14 +113,16 @@ public static class TraceProvider
 		/// </remarks>
 		public object? Parent
 		{
-			get => _parentRef?.Target;
+			get => _hardParentRef ?? _parentRef?.Target;
 			set
 			{
 				if (
 					!ReferenceEquals(_parentRef?.Target, value)
 					|| (_parentRef != null && value is null)
 				)
 				{
+					_hardParentRef = null;
```

Comment:
What if `Parent` is set while hard references are enabled?

Reply:
In the current context, it does not change anything, because changing the parent generally implies the element has been unloaded and the references has been removed. Yet, it makes sense to call disable instead.

Reply:
> In the current context, it does not change anything, because changing the parent generally implies the element has been unloaded and the references has been removed. 

Not necessarily, though - in some cases `Parent` gets set manually, usually in cases where the 'visual parent' is a native view and we want the 'logical parent' to be a managed view.

I guess this not a big deal though? It should not just fall back on the weak ref without functional implications if I understand correctly, and it's sufficiently uncommon that it shouldn't have any significant perf impact.